R/regularized_regression.R

```diff
@@ -409,3 +409,74 @@ mrash_weights <- function(X, y, init_prior_sd = TRUE, ...) {
   fit.mr.ash <- do.call("mr.ash", c(list(X = X, y = y, sa2 = if (init_prior_sd) init_prior_sd(X, y)^2 else NULL), args_list))
   predict(fit.mr.ash, type = "coefficients")[-1]
 }
+#' Extract Coefficients From Bayesian Linear Regression
+#'
+#' This function performs Bayesian linear regression using the `gbayes` function from 
+#' the `qgg` package. It then returns the estimated slopes.
+#'
+#' @param y A numeric vector of phenotypes.
+#' @param X A numeric matrix of genotypes.
+#' @param method A character string declaring the method/prior to be used. Options are 
+#' bayesN, bayesL, bayesA, bayesC, or bayesR.
+#' @param Z An optional numeric matrix of covariates.
+#' @return A vector containing the weights to be applied to each genotype in 
+#'   predicting the phenotype.
+#' @details This function fits a Bayesian linear regression model 
+#' @examples
+#' X <- matrix(rnorm(100000), nrow=1000)
+#' Z <- matrix(round(runif(3000, 0, 0.8), 0), nrow=1000)
+#' set1 <- sample(1:ncol(W), 5)
+#' set2 <- sample(1:ncol(W), 5)
+#' sets <- list(set1, set2)
+#' g <- rowSums(W[, c(set1, set2)])
+#' e <- rnorm(nrow(W), mean=0, sd=1)
+#' y <- g + e
+#' bayes_l_weights(y=y, X=X, Z=Z)
+#' bayes_r_weights(y=y, X=X, Z=Z)
+#' @importFrom qgg gbayes
+#' @export
+bayes_alphabet_weights <- function(y, X, method, Z=NULL){
+  # check for identical row lengths of response and genotype
+  if (!(length(y) == nrow(W))){
+    stop("All objects must have the same number of rows")
+  }
+  # check for identical row lengths of genotype and covariates
+  if (!is.null(X)){
+      if (nrow(X) != nrow(W)){
+        stop("Genotype and covariate matrices must have same number of rows")
+      }
+    }
+  
+  model = gbayes(y=y, 
+                 W=X,
+                 X=Z,
+                 method=method)
+  
+  return(model$bm)
```

Comment:
Double check that the return contains the same number of coef as you run other existing methods such as lasso -- i worry that it may or may not contain the intercept. Just make sure they are consistentn

Reply:
Yes, the returned vector has the same number of betas as there are snps